### PR TITLE
adl: init at 3.0.1

### PR DESCRIPTION
**Pull request by ghost**
###### Description of changes

https://github.com/RaitaroH/adl

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by WeebSorceress** [comment]
Once again, this was largely based on the flake pieces of @hlissner (https://github.com/hlissner/dotfiles/blob/master/packages/adl.nix) and @danielphan2003 (https://github.com/danielphan2003/flk/blob/main/pkgs/applications/video/adl/default.nix).

**Comment by WeebSorceress** [comment]
P.S. This package is "pending" as a **Draft** until `frece` gets merged: #179939

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -1038,6 +1038,8 @@ with pkgs;
     stdenv = gcc10StdenvCompat;
   };
 
+  adl = callPackage ../applications/video/adl { };
```
Comment (superherointj) [inline pkgs/top-level/all-packages.nix:1041]: You should have added `adl` to the applications (search `### APPLICATIONS`) section not tools. (And remember to keep alphabetical order.)

Comment (WeebSorceress) [reply pkgs/top-level/all-packages.nix]: Sorry, I didn't notice I did that...

**Comment by WeebSorceress** [comment]
> `./results/adl/bin/adl` doesn't show anything.

Should be something like this: https://asciinema.org/a/0ZGkTdpYtk0dbihEKkcxxszvN

But I think for it to work you should have logged in with [trackma](https://github.com/NixOS/nixpkgs/blob/2d48f0c18abe235f74679cd18b5e03adae4669af/pkgs/tools/misc/trackma/default.nix). That's why in my example it got pre-populated with those animes.

**Review by thiagokokada — APPROVED** [review]
Tested with my Anilist account, works fine.

**Pull request merged**